endpoint.cpp

```diff
@@ -344,6 +344,48 @@ void Endpoint::print_statistics()
     printf("\n}\n");
 }
 
+void Endpoint::untrim_msg(const struct buffer *from, struct buffer *to)
```

Comment:
But if you mark the message as trimmed you could just write zeros in the only place you need it. You don't need to copy the entire message.